sopel/modules/instagram.py

```diff
@@ -8,39 +8,30 @@
 """
 from __future__ import unicode_literals, absolute_import, print_function, division
 
-import re
 from datetime import datetime
 
 from requests import get
 
-from sopel import module, tools
+from sopel import module
 
 try:
     from ujson import loads
 except ImportError:
     from json import loads
 
 
-instagram_regex = r'.*(https?:\/\/(?:www\.){0,1}instagram\.com\/([a-zA-Z0-9_\.]{,30}\/)?p\/[a-zA-Z0-9_-]+)\s?.*'
-instagram_pattern = re.compile(instagram_regex)
+instagram_regex = r'(https?:\/\/(?:www\.){0,1}instagram\.com\/([a-zA-Z0-9_\.]{,30}\/)?p\/[a-zA-Z0-9_-]+)'
 
 
-def setup(bot):
-    if not bot.memory.contains('url_callbacks'):
-        bot.memory['url_callbacks'] = tools.SopelMemory()
-    bot.memory['url_callbacks'][instagram_pattern] = instaparse
-
-
-def shutdown(bot):
-    del bot.memory['url_callbacks'][instagram_pattern]
-
 # TODO: Parse Instagram profile page
 
 
-@module.rule(instagram_regex)
-def instaparse(bot, trigger):
+@module.url(instagram_regex)
+def instaparse(bot, trigger, match=None):
```

Comment:
The `match` parameter should be always provided when `@url` is used instead of `@rule`.

```suggestion
def instaparse(bot, trigger, match):
```